src/js/common/schemaform/FileField.jsx

```diff
@@ -0,0 +1,161 @@
+import { displayFileSize } from '../../common/utils/helpers';
+import environment from '../../common/helpers/environment';
+
+export default class FileField extends React.Component {
+  onAddFile = (e) => {
+    const files = this.props.formData || [];
+    this.props.onChange(files.concat({
+      uploading: true
+    }));
+    this.uploadFile(e.target.files[0])
+      .then(fileInfo => {
+        const uploadingFiles = this.props.formData;
+        this.props.onChange(_.set(uploadingFiles.length - 1, {
+          errorMessage: null,
+          uploading: false,
+          fileName: fileInfo.fileName,
+          maxSize: fileInfo.maxSize,
+          confirmationNumber: fileInfo.confirmationNumber
+        }, uploadingFiles));
+      })
+      .catch(message => {
+        const uploadingFiles = this.props.formData || [];
+        this.props.onBlur(this.props.idSchema.$id);
+        this.props.onChange(_.set(uploadingFiles.length - 1, {
+          uploading: false,
+          errorMessage: message
+        }, uploadingFiles));
+      });
+  }
+
+  uploadFile = (file) => {
+    const uiSchema = this.props.uiSchema;
+
+    if (file.size > uiSchema['ui:options'].maxSize) {
+      return Promise.reject('File is too large to be uploaded');
+    }
+
+    const payload = new FormData();
+    payload.append('upload', file);
+
+    return fetch(`${environment.API_URL}${uiSchema['ui:options'].endpoint}`, {
+      method: 'POST',
+      headers: {
+        'X-Key-Inflection': 'camel'
+      },
+      body: payload
+    }).then(resp => {
+      if (resp.ok) {
+        return resp.json();
+      }
+
+      return Promise.reject(resp.statusText);
+    }, error => {
+      return Promise.reject(error.message);
+    });
+  }
+
+  removeFile = (index) => {
+    const newFileList = this.props.formData.filter((file, idx) => index !== idx);
+    if (!newFileList.length) {
+      this.props.onChange();
+    }
+    this.props.onChange(newFileList);
+  }
+
+  render() {
+    const {
+      uiSchema,
+      errorSchema,
+      idSchema,
+      formData,
+      schema
+    } = this.props;
+
+    const uiOptions = uiSchema['ui:options'];
+    const files = formData || [];
+    const maxItems = schema.maxItems || Infinity;
+
+    const isUploading = files.some(file => file.uploading);
+
+    return (
+      <div>
+        {files.map((file, index) => {
+          const errors = _.get([index, '__errors'], errorSchema) || [];
+          return (
+            <ul key={index} className="schemaform-file-list">
+              {file.uploading && 'Uploading file...'}
+              {file.confirmationNumber &&
+                <li>{file.fileName} ({displayFileSize(file.maxSize)})&nbsp;
+                  <button
+                      onClick={() => this.removeFile(index)}
+                      className="usa-button usa-button-outline schemaform-file-remove-button"
+                      type="button">
+                    Remove
+                  </button>
+                </li>}
```

Comment:
Nit: Can we move that `}` down a line? I skipped right over it when I was reading this.